packages/inferno/src/DOM/events/delegation.ts

```diff
@@ -24,29 +24,53 @@ function getDelegatedEventObject(v) {
     onTouchStart: v
   };
 }
-const attachedEventCounts = getDelegatedEventObject(0);
-const attachedEvents = getDelegatedEventObject(null);
 
-export const delegatedEvents = getDelegatedEventObject(true);
+export function getDocumentId(doc: HTMLDocument) {
+  for (const docId in documentsScopes) {
+    if (documentsScopes[docId] === doc) {
+      return docId;
+    }
+  }
+
+  return null;
+}
+
+let uniqueId: number = 1;
 
-export function handleEvent(name: string, nextEvent: Function | LinkedEvent<any, any> | null, dom) {
+export function registerDocumentScope(doc: HTMLDocument) {
+  const docId = (uniqueId++).toString();
+  documentsScopes[docId] = doc;
+  attachedEventCounts[docId] = getDelegatedEventObject(0);
+  attachedEvents[docId] = getDelegatedEventObject(null);
+  delegatedEvents[docId] = getDelegatedEventObject(true);
```

Comment:
`delegatedEvents` does not need `docId`

Reply:
Do we actually need this docId at all? Cant we just pass doc to `handleEvent` too?

Reply:
delegatedEvents is now an object like it was before and removed it from this fn, we need docId only for collecting information to collections `attachedEventCounts` and `attachedEvents` that are used in `handleEvent` method